superset-frontend/src/views/CRUD/dataset/AddDatasetModal.tsx

```diff
@@ -97,8 +97,13 @@ const DatasetModal: FunctionComponent<DatasetModalProps> = ({
         onHide();
       })
       .catch(
-        createErrorHandler(errMsg =>
-          addDangerToast(t('Error while saving dataset: %s', errMsg)),
+        createErrorHandler((errMsg: unknown) =>
+          addDangerToast(
+            t(
+              'Error while saving dataset: %s',
+              (errMsg as { table_name?: string }).table_name,
```

Comment:
the error types are currently wrong. Error message can be either a string or an object, in this case the error message is scope to the input. Errors are most likely to occur in table_name field. 

Longer terms we should add error messages to the form so the errors appear next to the input that failed.

Reply:
Exactly!